docs/blockentities/index.md

```diff
@@ -1,141 +1,239 @@
+In both methods, it is important that you call super, as that adds basic information such as the position. The tag names `id`, `x`, `y`, `z`, `NeoForgeData` and `neoforge:attachments` are reserved by the super methods, and as such, you should not use them yourself.
 
-:::caution
-Synchronizing excessive/useless data for block entities can lead to network congestion. You should optimize your network usage by sending only the information the client needs when the client needs it. For instance, it is more often than not unnecessary to send the inventory of a block entity in the update tag, as this can be synchronized via its [`AbstractContainerMenu`][menu].
+:::info
+It is expected that Mojang will adapt the [Data Components][datacomponents] system to also work with block entities sometime during the next few updates. Once that happens, both saving to NBT and data attachments will be removed in favor of data components.
 :::
 
-### Synchronizing on Block Update
+Of course, you will want to set other values and not just work with defaults. You can do so freely, like with any other field. However, if you want the game to save those changes, you must call `#setChanged()` afterward, which marks the block entity's chunk as dirty (= in need of being saved). If you do not call that method, the block entity might get skipped during saving, as Minecraft's saving system only saves chunks that have been marked as dirty.
+
+## Tickers
 
-This method is a bit more complicated, but again you just need to override two or three methods. Here is a tiny example implementation of it:
+Another very common use of block entities, often in combination with some stored data, is ticking. Ticking means executing some code every game tick. This is done by overriding `EntityBlock#getTicker` and returning a `BlockEntityTicker`, which is basically a consumer with four arguments (level, position, blockstate and block entity), like so:
 
 ```java
-// In some subclass of BlockEntity
-@Override
-public CompoundTag getUpdateTag(HolderLookup.Provider registries) {
-  CompoundTag tag = new CompoundTag();
-  //Write your data into the tag
-  return tag;
+// Note: The ticker is defined in the block, not the block entity. However, it is good practice to
+// keep the ticking logic in the block entity in some way, for example by defining a static #tick method.
+public class MyEntityBlock extends Block implements EntityBlock {
+    // other stuff here
+
+    @Override
+    public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level level, BlockState state, BlockEntityType<T> type) {
+        // You can return different tickers here, depending on whatever factors you want. A common use case would be
+        // to return different tickers on the client or server, or only tick one side to begin with.
+        return type == MY_BLOCK_ENTITY.get() ? MyBlockEntity::tick : null;
```

Comment:
Due to the specific generic on the type and ticker, the ticker being returned must be casted to `BlockEntityTicker<T>` with an unchecked cast. This is why vanilla has `BaseEntityBlock.createTickerHelper()` which takes the incoming type, the actual type and the ticker and does this type comparison and cast for you. It's protected however, so one would need to copy or AT it as Neo currently does not provide an AT for it.